az-digital/az_quickstart
# Closes #559: Adds news feed importer migration(s).

## Description
WIP: Adds basic migration(s) for importing news stories from news.arizona.edu.

Had to make a multidev on news.arizona.edu that exposes the image `fid`s in order to complete the file/media migrations for the images:
https://azqs-559-uaqs-uanews.pantheonsite.io/feed/json/stories/campus-news

Still a lot left to do to match Quickstart 1 functionality:
- [x] Add configuration settings (and form) to allow news tags to be selected
- [x] Consider using migration groups (from Migrate Plus) to organize migrations and share config
- [x] Add event subscriber that alters migration source configs with correct import URLs based on config settings
- [x] Add mechanism for scheduling / importing stories periodically (possibly via https://www.drupal.org/project/migrate_queue_importer)
- [ ] Add migration(s) for external news stories (separate data source / field mappings)
- [x] Look into ways to simplify the files/media migrations
- [x] Replace Pantheon multidev URL with live news.arizona.edu URLs once `fid` is added to live news site views
- [x] Use dependency injection for all services.


## Related Issue




#559 

## How to test
Enable the az_news_feeds module
Navigate to /admin/content/az-news-feeds
Change URL
Navigate to /admin/config/migrate_queue_importer/cron_migration
Change cron frequency to 1 minute
Run Cron
Check content to see if it has been migrated.

## Types of changes


## Checklist:
